src/Uno.UI/UI/Xaml/Controls/Image/Image.cs

```diff
@@ -351,8 +355,10 @@ protected override Size MeasureOverride(Size availableSize)
 						case Stretch.Uniform:
 							var desiredSize = new Size();
 							var aspectRatio = sourceSize.Width / sourceSize.Height;
-							desiredSize.Width = Math.Min(knownWidth, knownHeight * aspectRatio);
-							desiredSize.Height = Math.Min(knownHeight, knownWidth / aspectRatio);
+							// Since apsect ratio can have a lot of decimal, iOS ceils Image size to 0.5 if it's not a precise size (like 111.111111111)
```

Comment:
Ceiling the aspect ratio to .5 seems rather radical, we're sure they're actually doing this? I mean 16:9 (1.7777777) and 16:10 (1.60000) would have the same value.

Reply:
It ceils the whole measure, not the AspectRatio